lib/plugins/acts_as_commentable/commentable_entity.rb

```diff
@@ -23,18 +23,42 @@ def count_all_comments
 
   # These below have all been redefined to work for the archive
 
-  # redefined for our archive to also not include
-  # hidden-by-admin comments.
-  # returns number of visible (not deleted) comments
-  def count_visible_comments
+  # The total number of visible comments on this commentable, not including
+  # deleted comments, spam comments, unreviewed comments, and comments hidden
+  # by an admin.
+  #
+  # This is the uncached version, and should only be used when calculating an
+  # accurate value is important (e.g. when updating the StatCounter in the
+  # database).
+  def count_visible_comments_uncached
     self.total_comments.where(
       hidden_by_admin: false,
       is_deleted: false,
       unreviewed: false,
       approved: true
     ).count
+  end
+
+  # The total number of visible comments on this commentable. Cached to reduce
+  # computation. The cache is manually expired whenever a comment is added,
+  # removed, or changes visibility, but the cache also expires after an hour in
+  # case of issues with the cache (e.g. stale data when calculating the count).
+  def count_visible_comments
+    @count_visible_comments ||=
+      Rails.cache.fetch(count_visible_comments_key, expires_in: 1.hour) do
```

Comment:
- Should we use `race_condition_ttl` here?
- Can you make this a config option?

Reply:
👍 

I set `race_condition_ttl` to 10 seconds, which seems to be the most common value for `race_condition_ttl` elsewhere in the codebase.